### ISSUE TITLE
False positive W0231 when using typing.overload.

### ISSUE THREAD

**Issue by mrshannon** [PyCQA/pylint#3020]
### Steps to reproduce
Minimum failing example.
```python
from typing import Union, overload


class Parent:

    def __init__(self, num: int) -> None:
        self.number = num


class Child(Parent):

    @overload
    def __init__(self, num: int) -> None:
        ...

    @overload
    def __init__(self, num: float) -> None:
        ...

    def __init__(self, num: Union[int, float]):
        super().__init__(round(num))

```
Run pylint on the above example code.

### Current behavior
```
pylint_init_bug.py:13:4: W0231: __init__ method from base class 'Parent' is not called (super-init-not-called)
pylint_init_bug.py:17:4: W0231: __init__ method from base class 'Parent' is not called (super-init-not-called)
```

### Expected behavior
The methods decorated with `typing.overload` should not be required to call the base `__init__` as they are "for the benefit of the type checker only" as indicated at [https://docs.python.org/3/library/typing.html#typing.overload](https://docs.python.org/3/library/typing.html#typing.overload).

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Apr  3 2019, 19:16:38) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

Same bug exists on the master branch which includes #2927.
```
pylint 2.4.0-dev0
astroid 2.3.0
Python 3.7.3 (default, Apr  3 2019, 19:16:38) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

**Comment by PCManticore** [comment]
Thanks, this definitely makes sense.

**Comment by gyermolenko** [comment]
I can try and take a look

**Comment by gyermolenko** [comment]
@mrshannon 
```
def __init__(self, num: Union[int, float]):
    super().__init__(round(num))
```
I think you don't need to put type annotations here, it was the sole purpose of overload methods to do that.
Also, for some time now It is not required to specify return type of `__init__`.

**Comment by mrshannon** [comment]
@gyermolenko 
You are correct about the return value of `__init__`.  However, the `Union` is required.  The CPython documentation for the typing module is incorrect in several areas (see https://github.com/python/mypy/issues/5466).

However, none of these changes prevent the original warnings from pylint.

**Comment by gyermolenko** [comment]
> However, none of these changes prevent the original warnings from pylint

Well, to fix the issue I've created PR :)

**Comment by mrshannon** [comment]
@gyermolenko 
I think I see now.  While the above is technically valid it's not required since `__init__` always has the same return type. My mistake
